pkg/types/chain_reader.go

```diff
@@ -44,16 +44,29 @@ type ChainReader interface {
 	// Note that implementations should ignore extra fields in params that are not expected in the call to allow easier
 	// use across chains and contract versions.
 	// Similarly, when using a struct for returnVal, fields in the return value that are not on-chain will not be set.
-	GetLatestValue(ctx context.Context, contractName, method string, params, returnVal any) error
+	GetLatestValue(ctx context.Context, contractName, method string, params, returnVal any, opts ...ReadOpt) error
```

Comment:
I'd rather go for something simpler like 
```suggestion
	GetLatestValue(ctx context.Context, singleIdentifier string, address string, params, returnVal any) error
```

Reply:
We need to consider the proto changes, at least alongside this API, but IMHO the protos should actually be what we think about first. Variadic options are just syntax sugar.

Reply:
So in this case, would it be better to alter the function signature in favor of explicit types instead of variadic options for the sake of keeping the proto API clean?

Reply:
I think they are separate concerns because protos cannot express functional options, they can only be used to transmit some resulting underlying config. IMHO it will be easier to think of that underlying proto first, and then after we can consider this kind of external API change. There are other options though, such as exposing (or matching) the Request/Request types generated by GRPC, which don't have the issues with ever-expanding method args.

Reply:
I'll make a new proposal.